internal/app/machined/pkg/system/services/extension.go

```diff
@@ -95,6 +95,35 @@ func (svc *Extension) DependsOn(r runtime.Runtime) []string {
 	return deps
 }
 
+// GetOCIOptions gets all OCI options from an Extension.
+func (svc *Extension) GetOCIOptions() []oci.SpecOpts {
+	ociOpts := []oci.SpecOpts{
+		oci.WithRootFSPath(filepath.Join(constants.ExtensionServicesRootfsPath, svc.Spec.Name)),
+		oci.WithRootFSReadonly(),
```

Comment:
can we also make this option configurable please?

so that we have `.container.security.writeableRootfs`: by default we enable `oci.WirthRootfsReadonly()`, but if configured, we skip this option